test/jit/test_symbolic_shape_analysis.py

```diff
@@ -141,6 +149,31 @@ def apply(fn):
             inputs[1].setType(inputs[1].type().with_sizes(size_2))
             torch._C._jit_pass_propagate_shapes_on_graph(t.graph)
             self.assertEqual(next(t.graph.outputs()).type().symbolic_sizes(), [4, 4, 8])
+            break
+
+    def test_binary_shape_fns_inplace(self):
+        def div_inplace_tensor(x: torch.Tensor, y: torch.Tensor):
+            z = x.div_(y)
+            return z
+
+        def add_inplace_tensor(x: torch.Tensor, y: torch.Tensor):
+            z = x.add_(y)
+            return z
+
+        binary_ops = [
+            div_inplace_tensor,
+            add_inplace_tensor,
+        ]
+
+        for fn in binary_ops:
+            size_1 = [4, 4, 8]  # x (can't broadcast because it's an inplace op)
+            size_2 = [4, 1, 8]
+            t = torch.jit.trace(fn, (torch.rand([4]), torch.rand([4])))
```

Comment:
Im pretty sure trace will replace things with their functional equivalent (or it sometimes does this). Can you script this instead ?